eland/tests/ml/test_imported_ml_model_pytest.py

```diff
@@ -268,24 +256,62 @@ def test_xgb_classifier(self, compress_model_definition, multi_class):
             overwrite=True,
             es_compress_model_definition=compress_model_definition,
         )
-        es_results = es_model.predict(test_data)
-
-        np.testing.assert_almost_equal(test_results, es_results, decimal=2)
+        # Get some test results
+        test_data = [[0.1, 0.2, 0.3, -0.5, 1.0], [1.6, 2.1, -10, 50, -1.0]]
+        check_prediction_equality(es_model, classifier, test_data)
 
         # Clean up
         es_model.delete_model()
 
     @requires_xgboost
-    @pytest.mark.parametrize("compress_model_definition", [True, False])
-    def test_xgb_regressor(self, compress_model_definition):
+    @pytest.mark.parametrize(
+        "objective", ["multi:softmax", "multi:softprob", "binary:logistic"]
+    )
+    @pytest.mark.parametrize("booster", ["gbtree", "dart"])
+    def test_xgb_classifier_objectives_and_booster(self, objective, booster):
+        # test both multiple and binary classification
+        if objective.startswith("multi"):
+            training_data = datasets.make_classification(
+                n_features=5, n_classes=3, n_informative=3
+            )
+            classifier = XGBClassifier(booster=booster, objective=objective)
+        else:
+            training_data = datasets.make_classification(n_features=5)
+            classifier = XGBClassifier(booster=booster, objective=objective)
+
         # Train model
-        training_data = datasets.make_regression(n_features=5)
-        regressor = XGBRegressor()
-        regressor.fit(training_data[0], training_data[1])
+        classifier.fit(training_data[0], training_data[1])
 
+        # Serialise the models to Elasticsearch
+        feature_names = ["f0", "f1", "f2", "f3", "f4"]
+        model_id = "test_xgb_classifier"
+
+        es_model = ImportedMLModel(
+            ES_TEST_CLIENT, model_id, classifier, feature_names, overwrite=True
+        )
         # Get some test results
         test_data = [[0.1, 0.2, 0.3, -0.5, 1.0], [1.6, 2.1, -10, 50, -1.0]]
-        test_results = regressor.predict(np.asarray(test_data))
+        check_prediction_equality(es_model, classifier, test_data)
+
+        # Clean up
+        es_model.delete_model()
+
+    @requires_xgboost
+    @pytest.mark.parametrize("compress_model_definition", [True, False])
+    @pytest.mark.parametrize(
+        "objective",
+        ["reg:squarederror", "reg:squaredlogerror", "reg:linear", "reg:logistic"],
```

Comment:
Should we add "reg:pseudohubererror" here too?

Reply:
@sethmlarson while we do support it, our current version of XGBoost does not include it :D

Reply:
XGBoost will include this in 1.2. I think we can add an automated test at that point.